lib/wallaroo/core/recovery/backend.pony

```diff
@@ -210,18 +212,23 @@ class FileBackend is Backend
         // Skip payload
         _file.seek(size.isize())
       | _LogCheckpointIdEntry =>
-        target_checkpoint_offset_end = _file.position() - 1
+        offset_end = _file.position() - 1
         r.append(_file.read(8))
         current_checkpoint_id = try r.u64_be()? else Fail(); 0 end
         if current_checkpoint_id == checkpoint_id then
+          // We want to find the last checkpoint in the log, because
```

Comment:
I think it should be an invariant that a checkpoint id never be repeated. If we abort a checkpoint, we should increment the checkpoint id.  Let's talk about this, as this seems to me to be masking a bug.